tracer/src/Datadog.Trace/DuckTyping/DuckType.Methods.cs

```diff
@@ -168,7 +168,7 @@ private static void CreateMethods(
                         DuckTypeTargetMethodNotFoundException.Throw(proxyMethodDefinition);
                     }
 
-                    targetMethod = targetMethod.MakeGenericMethod(proxyDuckAttribute.GenericParameterTypeNames.Select(name => Type.GetType(name, throwOnError: true)!).ToArray());
+                    targetMethod = targetMethod.MakeGenericMethod(proxyDuckAttribute.GenericParameterTypeNames.Select(name => GetTypeFromPartialName(name, throwOnError: true)!).ToArray());
```

Comment:
Do we _have_ to use it here? It complicates things because of the `throwOnError`...

Reply:
not for my case but the same could happen with `GenericParameterTypeNames` also declared in the DuckAttribute.

Reply:
yeah... we still don't really know _why_ it's necessary 😂